rootfs-version-check/module/rootfs-version-check

```diff
@@ -0,0 +1,90 @@
+#!/bin/sh
+
+set -ue
+
+# Is expected to contain two variable definitions with device paths:
+# - MENDER_ROOTFS_PART_A
+# - MENDER_ROOTFS_PART_B
+. /etc/mender/rootfs-version-check.conf
+
+MENDER_ROOTFS_PART_A_NUMBER="$(echo "$MENDER_ROOTFS_PART_A" | egrep -o '[0-9]+$')"
+MENDER_ROOTFS_PART_B_NUMBER="$(echo "$MENDER_ROOTFS_PART_B" | egrep -o '[0-9]+$')"
+
+active_num="$(fw_printenv mender_boot_part)"
+active_num="${active_num#mender_boot_part=}"
+if test $active_num -eq $MENDER_ROOTFS_PART_A_NUMBER; then
+    passive=$MENDER_ROOTFS_PART_B
+    passive_num=$MENDER_ROOTFS_PART_B_NUMBER
+else
+    passive=$MENDER_ROOTFS_PART_A
+    passive_num=$MENDER_ROOTFS_PART_A_NUMBER
+fi
+
+COMPARE_VERSIONS="/usr/share/utils/mender-compare-versions"
+case "$1" in
+    Download)
+        if [ ! -x "${COMPARE_VERSIONS}" ]; then
```

Comment:
This check should be repeated in `ArtifactCommit` to make sure you do not install an update which you later cannot move off of.

Reply:
How could that happen?

Reply:
I made this change as well.

Reply:
I meant only the check for the tool, not the version comparison. This could happen if the new image does not contain the tool. But doing both works too.